client.go

```diff
@@ -22,21 +23,32 @@ import (
 	"perun.network/go-perun/wallet"
 )
 
-// Client is a state channel client. It is the central controller to interact
-// with a state channel network. It can be used to propose channels to other
-// channel network peers.
-// ref https://pkg.go.dev/perun.network/go-perun/client?tab=doc#Client
-type Client struct {
-	cfg *Config
+type (
+	// Client is a state channel client. It is the central controller to interact
+	// with a state channel network. It can be used to propose channels to other
+	// channel network peers.
+	// ref https://pkg.go.dev/perun.network/go-perun/client?tab=doc#Client
+	Client struct {
+		cfg *Config
 
-	ethClient *ethclient.Client
-	client    *client.Client
+		ethClient *ethclient.Client
+		client    *client.Client
+		// mtx protects the acceptor
+		mtx      sync.RWMutex
+		acceptor ChannelAcceptor
 
-	w       *ethwallet.Wallet
-	onChain wallet.Account
+		w       *ethwallet.Wallet
+		onChain wallet.Account
 
-	dialer *net.Dialer
-}
+		dialer *net.Dialer
+	}
+
+	// ChannelAcceptor used to wrap a `func(*PaymentChannel)`
+	// function pointer for the `OnNewChannel` callback.
+	ChannelAcceptor interface {
+		Accept(*PaymentChannel)
+	}
```

Comment:
The naming is very confusing. This is not an `Acceptor` but only a callback/event consumer. There's nothing to accept here, the channel already got accepted and created. Please change this to something like
```suggestion
	// NewChannelCallback wraps a `func(*PaymentChannel)`
	// function pointer for the `Client.OnNewChannel` callback.
	NewChannelCallback interface {
		OnNew(*PaymentChannel)
	}
```
Also change the naming everywhere.